### PR TITLE
Added more instructions to OpenStack deploys

### PR DESCRIPTION
**Pull request by e-minguez**
* Removed the need to add a floating IP per 
https://github.com/openshift/installer/commit/d96e4c12bf096519cfa24075068d76a215286a3d
* Clarified more OSP requisites (RHCOS image, ssh key, external network 
name/id)
* Provided a clouds.yaml example file
* Added using /etc/hosts instead /etc/resolv.conf to access the API

### CODE REVIEW

**Comment by e-minguez** [comment]
I've updated the PR with @russellb suggestions (thanks!)

File: docs/user/openstack/README.md [hunk 1]
```diff
@@ -13,41 +13,84 @@ The installer assumes the following about the OpenStack cloud you run against:
...
   * `openstack object store account set --property Temp-URL-Key=superkey`
 
 * You may need to increase the security group related quotas from their default
-  values.
-  * For example: `openstack quota set --secgroups 100 --secgroup-rules 1000
-    openshift`
+  values. For example (as an OpenStack admin) `openstack quota set --secgroups 100 --secgroup-rules 1000 <project>`
+
+* The installer requires a proper RHCOS image in the OpenStack cluster or project:
+`openstack image create --container-format=bare --disk-format=raw --file redhat-coreos-${RHCOSVERSION}-openstack.raw redhat-coreos-${RHCOSVERSION}`
```
[the remark this answers is not in the document]
Comment (russellb) [reply docs/user/openstack/README.md]: OK, but the image is distributed as qcow2 right?  How about we keep this simple, and then maybe we can have a section with optimizations later like this.  We also aren't using Cinder here.

File: docs/user/openstack/README.md [hunk 2]
```diff
@@ -13,41 +13,84 @@ The installer assumes the following about the OpenStack cloud you run against:
...
 
-* `openstack floating ip create ${EXTERNAL_NETWORK}`
-* `openstack floating ip set --port lb-port ${FLOATING_IP_ADDRESS}`
+However, the installer will hang trying to reach the API as the node running the
+installer cannot resolve the service VM (the cluster should still come up
+successfully within the isolated network). As a temporary workaround:
 
-The service VM also hosts a DNS server that has enough records to bring up the
+* The service VM also hosts a DNS server that has enough records to bring up the
 cluster.  If you add the `${FLOATING_IP_ADDRESS}` as your first `nameserver`
```
[the remark this answers is not in the document]
Comment (russellb) [reply docs/user/openstack/README.md]: I meant drop it in favor of the content you added about just adding an /etc/hosts entry, which will get the same result

**Comment by russellb** [comment]
I'm fine with any more tweaks coming as follow-ups, this is a nice addition.  Thanks!

/lgtm

File: docs/user/openstack/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (e-minguez) [reply docs/user/openstack/README.md]: :+1:

File: docs/user/openstack/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (e-minguez) [reply docs/user/openstack/README.md]: :+1:

**Comment by e-minguez** [comment]
I've updated the PR with @russellb suggestions (thanks again!)

**Pull request merged**